src/temaWeek8Exception/main/Student.java

```diff
@@ -0,0 +1,160 @@
+package temaWeek8Exception.main;
+/**
+ * Class student with the following attributes:first name,
+ * last name, date of birth, gender and CNP.
+ */
+
+import java.time.Year;
+import java.util.Comparator;
+import java.util.Objects;
+
+public class Student {
+	
+	private final String firstName;
+	private final String lastName;
+	private final String dateOfBirth;
+	private final String gender;
+	private final String ID;
+	
+	//	class constructor
+	public Student(String firstName, String lastName, String dateOfBirth, String gender, String ID) {
+		this.firstName = firstName;
+		this.lastName = lastName;
+		this.dateOfBirth = dateOfBirth;
+		this.gender = gender;
+		this.ID = ID;
+	}
+	
+	//	nested class ExtendedException which creates custom exception extended from RuntimeException class
+	public static class ExtendedException extends RuntimeException {
+		public ExtendedException(String message) {
+			super(message);
+		}
+		
+	}
+	
+	//	 method to set firstName, throws IllegalArgumentException when has an illegal argument
+	public String addFirstName(String firstName) throws IllegalArgumentException {
+		
+		if (firstName==null || firstName.length()==0) {
+			throw new IllegalArgumentException("Invalid name!");
+		}
+		return firstName;
+	}
+	
+	//	 method to set lastName, throws IllegalArgumentException when has an illegal argument
+	public String addLastName(String lastName) throws IllegalArgumentException {
+		
+		if (lastName==null || lastName.length()==0) {
+			throw new IllegalArgumentException("Invalid name!");
+		}
+		return lastName;
+	}
+	
+	//	method to set dateOfBirth,throws NumberFormatException and ExtendedException
+	public String addDateOfBirth(String dateOfBirth) throws NumberFormatException, ExtendedException {
+		
+		final int y = Year.now().getValue() - 18;//calculates the year of birth as an int
+		
+		try {
+			int date = Integer.parseInt(dateOfBirth);//transforms method argument into an int
+			
+		} catch (NumberFormatException e) {        //catching the NumberFormatException when argument method is
+			System.err.println(e.getMessage());//not a numeric string type
+		}
+		int date = Integer.parseInt(dateOfBirth);
```

Comment:
You should throw an `IllegalArgumentException` otherwise it crashes the app as it is now (in case the `dateOfBirth` is not an integer).